### PR TITLE
Further improvements

### PR DESCRIPTION
**Pull request by RabeaGleissner**
@felipesere @jsuchy

This PR includes some small changes based on the feedback that Jim gave me during the IPM yesterday:
- makes naming of players more consistent
- removes duplication from tests
- makes test names clearer
- changes a long ternary into an if/else statement to improve readability 
- move `GameOverStub` to a separate file and remove duplication

There were a few things where I'm not sure how to change them. I'll make comments in the code. I'll speak to Jim to see if he has time to pair to work on these together.

### CODE REVIEW

File: src/main/java/de/rabea/game/Game.java [hunk 1]
```diff
@@ -26,12 +26,15 @@ public void play(Board board) {
...
+                currentPlayer = player2;
+            }
+            else currentPlayer = player1;
         }
     }
 
     private Board playOneRound(Player player, Board board) {
-        return board.placeMark(player.makeMove(board), player.mark());
+        return board.placeMark(player.getMove(board), player.mark());
     }
```
Comment (RabeaGleissner) [inline src/main/java/de/rabea/game/Game.java:38]: Jim's suggestion was that I shouldn't reach into the player twice here. And he said it seems strange to pass the board into `makeMove()` inside a method that is called on the board. I'm not sure how best to rectify that.

File: src/test/java/de/rabea/game/GameRunnerTest.java [hunk 2]
```diff
@@ -41,28 +38,22 @@ public void creates3x3Board() {
...
         GameRunner gameRunner = new GameRunner(new JavaFXUi(viewUpdaterSpy), new PlayerFactory(null));
         gameRunner.displayGameModeOptions();
+
         assertTrue(viewUpdaterSpy.hasShownGameModeOptions);
     }
 
     @Test
-    public void createsANewGame() {
+    public void createsNewGameWithGuiPlayerAndUnbeatablePlayer() {
         GameRunner gameRunner = new GameRunner(new JavaFXUi(viewUpdaterSpy), new PlayerFactory(null));
```
Comment (RabeaGleissner) [inline src/test/java/de/rabea/game/GameRunnerTest.java:55]: What I really want to test here is that it creates a new Game. That's it. But I don't know how to assert on that. Options I'm seeing:
- asserting that game != null (bad... I've been told before never to assert on this)
- check that the game has the correct players (which I'm doing now - but that's not really what I want to test)

File: src/test/java/de/rabea/game/GameTest.java [hunk 3]
```diff
@@ -19,9 +19,9 @@ public void setup() {
...
 
     @Test
-    public void playsHuman3x3GameOnce() {
-        FakeHumanPlayer fakePlayer1 = new FakeHumanPlayer(fakeConsoleUI, X);
-        FakeHumanPlayer fakePlayer2= new FakeHumanPlayer(fakeConsoleUI, O);
+    public void playsHumanGameOnce() {
+        FakeConsolePlayer fakePlayer1 = new FakeConsolePlayer(fakeConsoleUI, X);
+        FakeConsolePlayer fakePlayer2= new FakeConsolePlayer(fakeConsoleUI, O);
         fakePlayer1.futureMoves(0,1,2);
         fakePlayer2.futureMoves(4,5);
```
Comment (RabeaGleissner) [inline src/test/java/de/rabea/game/GameTest.java]: Jim's comment was that these numbers (for future moves) seem random. How can I improve this? I could hide them inside a method with a more descriptive name for example?

File: src/main/java/de/rabea/game/Game.java (hunk 1 above)
Comment (felipesere) [inline src/main/java/de/rabea/game/Game.java:38]: You could flip this on its head:
`return player.makeMoveOn(board)`

And internally that uses `board.placeMark(someRandomMove, this.name)`

File: src/test/java/de/rabea/game/GameRunnerTest.java (hunk 2 above)
Comment (felipesere) [inline src/test/java/de/rabea/game/GameRunnerTest.java:55]: This is a tough one!
Checking for `x != null` makes sense IF there is a situation where `x == null` will hold.
Testing that you get something back that _a kind of Game_ is already guaranteed by the compiler.
Now, testing that the players that get back are of the right _type_, feels like I am testing the _factory_. 
I am asserting that `GameMode.GuiHumanVsComputer` gives me a `HumanPlayer` and a `UnbeatableComputerPlayer`, right?

I am not really sure how to test that 'a correct game is constructed' though...

File: src/test/java/de/rabea/game/GameTest.java (hunk 3 above)
Comment (felipesere) [inline src/test/java/de/rabea/game/GameTest.java]: Hiding it in a method is a band aid.
Maybe you can work on making the numbers more descriptive?
Or the method name? And with "more descriptive" I don't mean longer, but just more precise.
Maybe `fakePlayer1.willMakeMoves(TOP_LEFT, TOP_CENTER, TOP_RIGHT)`?
Or `fakePlayer1.willPlayOn(TOP_ROW)`?

File: src/main/java/de/rabea/game/Game.java (hunk 1 above)
Comment (RabeaGleissner) [inline src/main/java/de/rabea/game/Game.java:38]: Ah yes! I like that. Done!

File: src/test/java/de/rabea/game/GameTest.java (hunk 3 above)
Comment (RabeaGleissner) [inline src/test/java/de/rabea/game/GameTest.java]: Thanks for that! Done!

**Pull request merged**